TLM/TLM/Custom/PathFinding/CustomPathManager.cs

```diff
@@ -28,22 +25,70 @@ public class CustomPathManager : PathManager {
 
         private CustomPathFind[] _replacementPathFinds;
 
-        public static CustomPathManager _instance;
+        public static CustomPathManager _instance => PathManager.instance as CustomPathManager;
+
+        private PathManager stockPathManager_;
+
+        private static FastList<ISimulationManager> GetSimulationManagers() =>
+            typeof(SimulationManager)
+            .GetField("m_managers", BindingFlags.Static | BindingFlags.NonPublic)
+            ?.GetValue(null)
+            as FastList<ISimulationManager>
+            ?? throw new Exception("could not get SimulationManager.m_managers");
+
+        private static FieldInfo PathManagerInstance =>
+            typeof(Singleton<PathManager>)
+            .GetField(
+            "sInstance",
+            BindingFlags.Static | BindingFlags.NonPublic) ??
+            throw new Exception("pathManagerInstance is null");
 
 #if QUEUEDSTATS
-        public static uint TotalQueuedPathFinds {
-            get; private set;
-        }
+        public static uint TotalQueuedPathFinds { get; private set; }
 #endif
 
-        public static bool InitDone {
-            get; private set;
+        public static void OnLevelLoaded() {
+            try {
+                Log.Info("CustomPathManager.OnLevelLoaded() called.");
+                PathManager.instance.gameObject.AddComponent<CustomPathManager>();
+                Log._Debug("Added CustomPathManager to gameObject List");
+            } catch(Exception ex) {
+                string error =
+                    "Traffic Manager: President Edition failed to load. You can continue " +
+                    "playing but it's NOT recommended. Traffic Manager will not work as expected.";
+                Log.Error(error);
+                Log.Error($"Path manager replacement error: {ex}");
+                UIView.library
+                        .ShowModal<ExceptionPanel>(
+                            "ExceptionPanel")
+                        .SetMessage(
+                            "TM:PE failed to load",
+                            error,
+                            true);
+            }
         }
 
-        // On waking up, replace the stock pathfinders with the custom one
         [UsedImplicitly]
-        public new virtual void Awake() {
-            _instance = this;
+        protected override void Awake() {
+            Log._Debug("Waking up CustomPathManager.");
+
+            // On waking up, replace the stock pathfinders with the custom one
+            // but retain the original version for future replace
+            // also suppress call to base class.
+            stockPathManager_ = PathManager.instance
+                ?? throw new Exception("stockPathManager is null");
+            Log._Debug($"Got stock PathManager instance {stockPathManager_?.GetName()}");
+            PathManagerInstance.SetValue(null, this);
+            Log._Debug("Should be custom: " + PathManager.instance.GetType());
```

Comment:
Can this fail? Do we need to log this every time?

Reply:
I just copied it over. it's a debug log so it's not every time. should I delete it?

Reply:
it can be confusing to track CustomPathManger and stock PathManager. and something might fail during hot-reload or something.